src/widgets/SequentialSteps.js

```diff
@@ -0,0 +1,159 @@
+/* eslint-disable react/forbid-prop-types */
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Ionicons from 'react-native-vector-icons/Ionicons';
+import FontAwesome from 'react-native-vector-icons/FontAwesome';
+
+import { StyleSheet, View, Text, TouchableOpacity } from 'react-native';
+
+import { WARM_GREY, SUSSOL_ORANGE } from '../globalStyles';
+
+/**
+ * Component which renders either a pressable text field for
+ * triggering an AutoCompleteSelector modal, or an input field
+ * for a TextEditor modal. Primary use for within the ByProgramModal,
+ * but will display a selection of choices for the user to step through
+ * sequentially, disabling all steps after the current step in the
+ * sequence. An error is used as a fallback for callers to disable
+ * a step in the sequence and display an error message.
+ *
+ * @prop {Func} onPress       onPress handler for each step object, returns {key, field, isLastStep}
+ * @prop {array} steps        An array of step objects, example below*
+ * @prop {number} currentStep The index of the current step in the sequence
+ * @
+ *
+ */
+export default class SequentialSteps extends React.PureComponent {
+  renderStepIcon = ({ isCurrentStep, isLessThanCurrentStep, step, complete }) => {
+    const { error } = step;
+    return (
+      <View style={{ minWidth: 40 }}>
+        {isCurrentStep && error && <Ionicons name="ios-close" color="red" size={30} />}
+        {((isLessThanCurrentStep && !error) || complete) && (
+          <Ionicons name="md-checkmark" color="green" size={30} />
+        )}
+        {isCurrentStep && !error && (
+          <Ionicons name="md-arrow-round-forward" color={SUSSOL_ORANGE} size={30} />
+        )}
+      </View>
+    );
+  };
+
+  renderInputField = ({ canEdit, step }) => {
+    const { textContent, text } = localStyles;
+    const { name, placeholder } = step;
+    const textStyle = { ...text, color: canEdit ? 'white' : WARM_GREY, marginBottom: 10 };
+    const containerStyle = { borderBottomColor: 'white', borderBottomWidth: 1, ...textContent };
+    return (
+      <View style={containerStyle}>
+        <Text numberOfLines={1} ellipsizeMode="tail" style={textStyle}>
+          {name || placeholder}
```

Comment:
@joshxg, was wondering if we should display placeholder if !isLessThanCurrentStep

Reply:
I thought the same trying to think of the best way to give feedback to the user on what they should be doing. 

Currently there is an arrow pointing to the step which is the 'current step', and the sequential steps have a lower opacity. When steps are complete they get a tick and their opacity is kept high. Not completely happy, but thought this would be sufficient.

I also couldn't think of a good placeholder
- 'Please do the step above first' (3 times)
- 'Please do X step' 3 times 
- blank/nothing

Open to suggestions!

Keeping them visible also kinda gives feedback to what is 'next' to do